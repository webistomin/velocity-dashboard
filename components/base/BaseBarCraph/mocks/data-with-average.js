export const barGraphDataWithAverage = {
  labels: ['Mon', 'Tue', 'Fri', 'Thu', 'Wed', 'Sat', 'Sun'],
  datasets: [
    {
      label: 'Average',
      borderColor: '#F7C137',
      fill: false,
      lineTension: 0,
      data: [110, 80, 100, 110, 74, 55, 124],
      type: 'line',
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Comfort',
      backgroundColor: '#2E5BFF',
      data: [82, 65, 75, 130, 55, 62, 120],
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Premium',
      backgroundColor: '#8C54FF',
      data: [90, 75, 80, 60, 45, 117],
    },
  ],
};
