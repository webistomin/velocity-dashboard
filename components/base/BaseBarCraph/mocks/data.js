export const barGraphData = {
  labels: ['April', 'May', 'June'],
  datasets: [
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Comfort',
      lineTension: 0.4,
      backgroundColor: '#2E5BFF',
      data: [60, 65, 85],
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Premium',
      backgroundColor: '#8C54FF',
      data: [70, 80, 90],
    },
  ],
};
