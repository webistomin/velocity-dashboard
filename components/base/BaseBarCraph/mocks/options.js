export const options = {
  responsive: true,
  maintainAspectRatio: false,
  scales: {
    yAxes: [
      {
        gridLines: {
          color: '#B0BAC9',
          borderDash: [3, 3],
        },
        ticks: {
          suggestedMin: 0,
          suggestedMax: 125,
          fontColor: '#B0BAC9',
        },
      },
    ],
    xAxes: [
      {
        gridLines: {
          color: '#B0BAC9',
          borderDash: [3, 3],
        },
        ticks: {
          fontColor: '#B0BAC9',
        },
      },
    ],
  },
  legend: {
    display: true,
    align: 'end',
    labels: {
      boxWidth: 10,
      usePointStyle: true,
      padding: 20,
      fontColor: '#B0BAC9',
      fontFamily: 'Rubik, system-ui, sans-serif',
    },
  },
  elements: {
    point: {
      radius: 5,
      hoverRadius: 4,
      borderWidth: 3,
      hoverBorderWidth: 3,
      backgroundColor: '#FFF',
    },
  },
};
