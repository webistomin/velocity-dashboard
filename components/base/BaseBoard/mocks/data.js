export const boardData = [
  {
    title: 'Backlog',
    color: 'blue',
    tasks: [
      {
        title: 'Spire1',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire2',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire3',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'In progress',
    color: 'purple',
    tasks: [
      {
        title: 'Spire5',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire6',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire7',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire8',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Validate',
    color: 'darkturquoise',
    tasks: [
      {
        title: 'Spire9',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire10',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire11',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire12',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Complete',
    color: 'yellow',
    tasks: [
      {
        title: 'Spire13',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire14',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire15',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire16',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
];
