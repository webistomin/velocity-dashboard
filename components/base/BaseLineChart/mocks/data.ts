import { ChartData } from 'chart.js';

export const lineGraphData: ChartData = {
  labels: [
    '',
    'Mar 1',
    'Mar 8',
    'Mar 15',
    'Mar 22',
    'Mar 29',
    'Apr 5',
    'Apr 12',
    'Apr 19',
    'Apr 26',
    'May 2',
    'May 9',
    'May 16',
    'May 23',
    'May 30',
    'Jun 5',
    'Jun 10',
    'Jun 19',
  ],
  datasets: [
    {
      label: 'Revenue',
      borderColor: '#2E5BFF',
      pointBackgroundColor: '#FFF',
      fill: true,
      lineTension: 0,
      data: [
        {
          x: 0,
          y: 8000,
        },
        {
          x: 1,
          y: 11500,
        },
        {
          x: 2,
          y: 9800,
        },
        {
          x: 3,
          y: 15100,
        },
        {
          x: 4,
          y: 23000,
        },
        {
          x: 5,
          y: 22000,
        },
        {
          x: 6,
          y: 16000,
        },
        {
          x: 7,
          y: 13800,
        },
        {
          x: 8,
          y: 16300,
        },
        {
          x: 9,
          y: 17000,
        },
        {
          x: 10,
          y: 24500,
        },
        {
          x: 11,
          y: 19000,
        },
        {
          x: 12,
          y: 12000,
        },
        {
          x: 13,
          y: 8900,
        },
        {
          x: 14,
          y: 16000,
        },
        {
          x: 15,
          y: 27000,
        },
        {
          x: 16,
          y: 22000,
        },
        {
          x: 17,
          y: 21000,
        },
      ],
    },
    {
      label: 'Trips',
      borderColor: '#8C54FF',
      pointBackgroundColor: '#FFF',
      fill: true,
      lineTension: 0,
      data: [
        {
          x: 0,
          y: 4800,
        },
        {
          x: 1,
          y: 5000,
        },
        {
          x: 2,
          y: 7500,
        },
        {
          x: 3,
          y: 7000,
        },
        {
          x: 4,
          y: 11500,
        },
        {
          x: 5,
          y: 13000,
        },
        {
          x: 6,
          y: 7500,
        },
        {
          x: 7,
          y: 8500,
        },
        {
          x: 8,
          y: 6000,
        },
        {
          y: 13700,
        },
        {
          x: 10,
          y: 18000,
        },
        {
          x: 11,
          y: 17600,
        },
        {
          x: 12,
          y: 2500,
        },
        {
          x: 13,
          y: 6000,
        },
        {
          x: 14,
          y: 9000,
        },
        {
          x: 15,
          y: 22500,
        },
        {
          x: 16,
          y: 14000,
        },
        {
          x: 17,
          y: 19000,
        },
      ],
    },
  ],
};
