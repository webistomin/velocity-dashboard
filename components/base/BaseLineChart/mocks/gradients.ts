export const gradients = [
  {
    red: 46,
    green: 91,
    blue: 255,
  },
  {
    red: 140,
    green: 84,
    blue: 255,
  },
];
