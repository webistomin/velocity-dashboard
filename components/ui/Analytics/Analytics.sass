+b('analytics')
  +e('grid')
    display: grid
    justify-content: start
    grid-template-columns: repeat(12, 1fr)
    grid-gap: rem(14px)

    +media($screen-sm)
      grid-gap: rem(24px)

    +media($screen-lg)
      grid-gap: rem(34px)

  +e('block')
    &:first-of-type
      grid-column: 12 span
      grid-row: 4 span

    &:nth-of-type(2),
    &:nth-of-type(3),
    &:nth-of-type(4),
    &:nth-of-type(5)
      grid-column: 12 span
      grid-row: 1 span

    &:nth-of-type(6)
      grid-column: 12 span
      grid-row: 2 span

    &:nth-of-type(7)
      grid-column: 12 span
      grid-row: 3 span

    +media($screen-sm)
      &:nth-of-type(2),
      &:nth-of-type(3),
      &:nth-of-type(4),
      &:nth-of-type(5)
        grid-column: 6 span
        grid-row: 1 span

    +media($screen-lg)
      &:nth-of-type(2),
      &:nth-of-type(3),
      &:nth-of-type(4),
      &:nth-of-type(5)
        grid-column: 3 span
        grid-row: 1 span

      &:nth-of-type(6)
        grid-column: 4 span
        grid-row: 3 span

      &:nth-of-type(7)
        grid-column: 8 span
        grid-row: 3 span


