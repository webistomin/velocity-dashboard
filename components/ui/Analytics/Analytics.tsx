import { VueComponent } from 'types/vue-components';
import { Component, Prop } from 'nuxt-property-decorator';
import { VNode } from 'vue';
import { ChartData, ChartOptions } from 'chart.js';
import formatDistance from 'date-fns/formatDistance';
// @ts-ignore
import LazyHydrate from 'vue-lazy-hydration';

import BaseBlock from 'components/base/BaseBlock';
import BaseLineChart from 'components/base/BaseLineChart';
import BaseStat from 'components/base/BaseStat';
import BaseList from 'components/base/BaseList';
import BaseBarGraph from 'components/base/BaseBarCraph';
import { IPageAnalytics } from 'common/types/pages/analytics';
import { IBaseList } from 'components/base/BaseList/BaseList';
import BaseEmpty from 'components/base/BaseEmpty';

import './Analytics.sass';

export interface IAnalyticsPageProps {
  content: IPageAnalytics | null;
}

@Component({
  name: 'Analytics',
  components: { LazyHydrate },
})
export default class Analytics extends VueComponent<IAnalyticsPageProps> {
  @Prop({ required: true })
  private readonly content!: IAnalyticsPageProps['content'];

  public lineOptions: ChartOptions = {
    responsive: true,
    maintainAspectRatio: false,
    scales: {
      yAxes: [
        {
          gridLines: {
            color: '#B0BAC9',
            borderDash: [3, 3],
          },
          ticks: {
            suggestedMin: 0,
            suggestedMax: 125,
            fontColor: '#B0BAC9',
            callback: (value) => {
              let tempVal = value.toString();
              if (tempVal.length >= 4) tempVal = tempVal.substring(0, tempVal.length - 3) + 'k ';
              return '$' + tempVal;
            },
          },
        },
      ],
      xAxes: [
        {
          gridLines: {
            color: '#B0BAC9',
            borderDash: [3, 3],
          },
          ticks: {
            fontColor: '#B0BAC9',
          },
        },
      ],
    },
    legend: {
      display: true,
      align: 'end',
      labels: {
        boxWidth: 10,
        usePointStyle: true,
        padding: 20,
        fontColor: '#B0BAC9',
      },
    },
    elements: {
      point: {
        radius: 5,
        hoverRadius: 4,
        borderWidth: 3,
        hoverBorderWidth: 3,
      },
    },
  };

  public mixedOptions: ChartOptions = {
    responsive: true,
    maintainAspectRatio: false,
    scales: {
      yAxes: [
        {
          gridLines: {
            color: '#B0BAC9',
            borderDash: [3, 3],
          },
          ticks: {
            suggestedMin: 0,
            suggestedMax: 125,
            fontColor: '#B0BAC9',
          },
        },
      ],
      xAxes: [
        {
          gridLines: {
            color: '#B0BAC9',
            borderDash: [3, 3],
          },
          ticks: {
            fontColor: '#B0BAC9',
          },
        },
      ],
    },
    legend: {
      display: true,
      align: 'end',
      labels: {
        boxWidth: 10,
        usePointStyle: true,
        padding: 20,
        fontColor: '#B0BAC9',
      },
    },
    elements: {
      point: {
        radius: 5,
        hoverRadius: 4,
        borderWidth: 3,
        hoverBorderWidth: 3,
        backgroundColor: '#FFF',
      },
    },
  };

  public get getRevenueChartData(): ChartData | undefined {
    const revenue = this.content?.revenue;

    if (!revenue) {
      return undefined;
    }

    return {
      labels: revenue.labels,
      datasets: [
        {
          label: 'Revenue',
          borderColor: '#2E5BFF',
          pointBackgroundColor: '#FFF',
          fill: true,
          lineTension: 0,
          data: revenue.data[0],
        },
        {
          label: 'Trips',
          borderColor: '#8C54FF',
          pointBackgroundColor: '#FFF',
          fill: true,
          lineTension: 0,
          data: revenue.data[1],
        },
      ],
    };
  }

  public get getLatestTrips(): IBaseList[] | undefined {
    return this.content?.latestTrips?.trips?.map((trip) => {
      return {
        image: '/img/avatar-placeholder.svg',
        name: `${trip.trip.startAddress} ${trip.trip.endAddress}`,
        text: `${trip.trip.endTime ? formatDistance(new Date(trip.trip.endTime), new Date()) : ''}`,
        statistic: `${trip.trip.distance} miles`,
        price: `$${trip.trip.price}`,
      };
    });
  }

  public get getTripsByWeekdayChartData(): ChartData | undefined {
    const tripsByWeekday = this.content?.tripsByWeekday;

    if (!tripsByWeekday) {
      return undefined;
    }

    return {
      labels: ['Mon', 'Tue', 'Fri', 'Thu', 'Wed', 'Sat', 'Sun'],
      datasets: [
        {
          label: 'Average',
          borderColor: '#F7C137',
          fill: false,
          lineTension: 0,
          data: tripsByWeekday.datasets[0],
          type: 'line',
        },
        {
          barPercentage: 0.5,
          categoryPercentage: 0.3,
          label: 'Comfort',
          backgroundColor: '#2E5BFF',
          data: tripsByWeekday.datasets[1],
        },
        {
          barPercentage: 0.5,
          categoryPercentage: 0.3,
          label: 'Premium',
          backgroundColor: '#8C54FF',
          data: tripsByWeekday.datasets[2],
        },
      ],
    };
  }

  public render(): VNode {
    return (
      <section class='analytics'>
        <div class='container'>
          <div class='analytics__grid'>
            <BaseBlock class='analytics__block' title='Revenue'>
              {this.getRevenueChartData ? (
                <BaseLineChart
                  chartData={this.getRevenueChartData}
                  options={this.lineOptions}
                  gradients={[
                    {
                      red: 46,
                      green: 91,
                      blue: 255,
                    },
                    {
                      red: 140,
                      green: 84,
                      blue: 255,
                    },
                  ]}
                />
              ) : (
                <BaseEmpty />
              )}
            </BaseBlock>

            <BaseBlock class='analytics__block' title='Vehicles on track'>
              {this.content?.vehiclesOnTrack ? (
                <BaseStat
                  value={this.content.vehiclesOnTrack.todayCount}
                  prevValue={this.content.vehiclesOnTrack.yesterdayCount}
                  measure='$'
                  icon='icon-check'
                  color='green'
                  align='row'
                />
              ) : (
                <BaseEmpty />
              )}
            </BaseBlock>

            <BaseBlock class='analytics__block' title='Distance driven'>
              {this.content?.distanceDriven ? (
                <BaseStat
                  value={this.content.distanceDriven.todayCount}
                  prevValue={this.content.distanceDriven.yesterdayCount}
                  measure='mi'
                  icon='icon-pin'
                  color='blue'
                  align='row'
                />
              ) : (
                <BaseEmpty />
              )}
            </BaseBlock>

            <BaseBlock class='analytics__block' title='Energy consumed'>
              {this.content?.energyConsumed ? (
                <BaseStat
                  value={this.content.energyConsumed.todayCount}
                  prevValue={this.content.energyConsumed.yesterdayCount}
                  measure='kWh'
                  icon='icon-lightning'
                  color='purple'
                  align='row'
                />
              ) : (
                <BaseEmpty />
              )}
            </BaseBlock>

            <BaseBlock class='analytics__block' title='Total drive time'>
              {this.content?.totalDriveTime ? (
                <BaseStat
                  value={this.content.totalDriveTime.todayCount}
                  prevValue={this.content.totalDriveTime.yesterdayCount}
                  measure='h'
                  icon='icon-graph'
                  color='yellow'
                  align='row'
                />
              ) : (
                <BaseEmpty />
              )}
            </BaseBlock>

            <LazyHydrate when-visible>
              <BaseBlock class='analytics__block' title='Latest trips'>
                {this.getLatestTrips ? <BaseList list={this.getLatestTrips} /> : <BaseEmpty />}
              </BaseBlock>
            </LazyHydrate>

            <LazyHydrate when-visible>
              <BaseBlock class='analytics__block' title='Trips by weekday'>
                {this.getTripsByWeekdayChartData ? (
                  <BaseBarGraph chartData={this.getTripsByWeekdayChartData} options={this.mixedOptions} />
                ) : (
                  <BaseEmpty />
                )}
              </BaseBlock>
            </LazyHydrate>
          </div>
        </div>
      </section>
    );
  }
}
