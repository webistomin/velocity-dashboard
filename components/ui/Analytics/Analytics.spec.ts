import { shallowMount } from '@vue/test-utils';
import { ANALYTICS_PAGE_MOCKED_RESPONSE } from 'server/mocks/analytics-page';
import Analytics from './Analytics';

describe('Analytics UI', () => {
  const BaseBarGraphSelector = 'basebargraph-stub';
  const BaseListSelector = 'baselist-stub';

  const factory = (propsData = {}, options = {}) => {
    return shallowMount(Analytics, {
      propsData: {
        content: {
          ...ANALYTICS_PAGE_MOCKED_RESPONSE(),
          vehiclesOnTrack: {
            todayCount: 1,
          },
          distanceDriven: {
            todayCount: 1,
          },
          energyConsumed: {
            todayCount: 1,
          },
          totalDriveTime: {
            todayCount: 1,
          },
        },
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Return undefined if revenue is not provided', () => {
    const wrapper = factory({
      content: {
        ...ANALYTICS_PAGE_MOCKED_RESPONSE(),
        revenue: undefined,
      },
    });
    const revenue = wrapper.vm.getRevenueChartData;
    expect(revenue).toEqual(undefined);
  });

  it('Return undefined if trips by weekday are not provided', () => {
    const wrapper = factory({
      content: {
        ...ANALYTICS_PAGE_MOCKED_RESPONSE(),
        tripsByWeekday: undefined,
      },
    });
    const revenue = wrapper.vm.getTripsByWeekdayChartData;
    expect(revenue).toEqual(undefined);
  });

  it('Hide BaseBarGraph if trips by weekday are not provided', () => {
    const wrapper = factory({
      content: {
        ...ANALYTICS_PAGE_MOCKED_RESPONSE(),
        tripsByWeekday: undefined,
      },
    });
    const bar = wrapper.find(BaseBarGraphSelector);
    expect(bar.exists()).toBe(false);
  });

  // it('Show BaseBarGraph if  trips by weekday are provided', () => {
  //   const wrapper = factory();
  //   const bar = wrapper.find(BaseBarGraphSelector);
  //   expect(bar.exists()).toBe(true);
  // });

  it('Hide BaseList with trips if latest trips are not provided', () => {
    const wrapper = factory({
      content: {
        ...ANALYTICS_PAGE_MOCKED_RESPONSE(),
        latestTrips: undefined,
      },
    });
    const list = wrapper.find(BaseListSelector);
    expect(list.exists()).toBe(false);
  });

  // it('Show BaseList with trips if latest trips are provided', () => {
  //   const wrapper = factory();
  //   const list = wrapper.find(BaseListSelector);
  //   expect(list.exists()).toBe(true);
  // });
});
