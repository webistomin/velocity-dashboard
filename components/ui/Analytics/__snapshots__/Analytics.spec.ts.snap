// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics UI Matches snapshot 1`] = `
"<section class=\\"analytics\\">
  <div class=\\"container\\">
    <div class=\\"analytics__grid\\">
      <baseblock-stub title=\\"Revenue\\" class=\\"analytics__block\\">
        <baselinechart-stub chartid=\\"line-chart\\" width=\\"400\\" height=\\"400\\" cssclasses=\\"\\" plugins=\\"\\" chartdata=\\"[object Object]\\" options=\\"[object Object]\\" gradients=\\"[object Object],[object Object]\\" fallbacktext=\\"Linear graph\\"></baselinechart-stub>
      </baseblock-stub>
      <baseblock-stub title=\\"Vehicles on track\\" class=\\"analytics__block\\">
        <basestat-stub value=\\"1\\" prevvalue=\\"0\\" measure=\\"$\\" icon=\\"icon-check\\" color=\\"green\\" align=\\"row\\"></basestat-stub>
      </baseblock-stub>
      <baseblock-stub title=\\"Distance driven\\" class=\\"analytics__block\\">
        <basestat-stub value=\\"1\\" prevvalue=\\"0\\" measure=\\"mi\\" icon=\\"icon-pin\\" color=\\"blue\\" align=\\"row\\"></basestat-stub>
      </baseblock-stub>
      <baseblock-stub title=\\"Energy consumed\\" class=\\"analytics__block\\">
        <basestat-stub value=\\"1\\" prevvalue=\\"0\\" measure=\\"kWh\\" icon=\\"icon-lightning\\" color=\\"purple\\" align=\\"row\\"></basestat-stub>
      </baseblock-stub>
      <baseblock-stub title=\\"Total drive time\\" class=\\"analytics__block\\">
        <basestat-stub value=\\"1\\" prevvalue=\\"0\\" measure=\\"h\\" icon=\\"icon-graph\\" color=\\"yellow\\" align=\\"row\\"></basestat-stub>
      </baseblock-stub>
      <anonymous-stub idletimeout=\\"2000\\" whenvisible=\\"true\\">
        <baseblock-stub title=\\"Latest trips\\" class=\\"analytics__block\\">
          <baselist-stub list=\\"[object Object],[object Object]\\" limit=\\"6\\"></baselist-stub>
        </baseblock-stub>
      </anonymous-stub>
      <anonymous-stub idletimeout=\\"2000\\" whenvisible=\\"true\\">
        <baseblock-stub title=\\"Trips by weekday\\" class=\\"analytics__block\\">
          <basebargraph-stub chartid=\\"bar-chart\\" width=\\"400\\" height=\\"400\\" cssclasses=\\"\\" plugins=\\"\\" chartdata=\\"[object Object]\\" options=\\"[object Object]\\" fallbacktext=\\"Bar graph\\"></basebargraph-stub>
        </baseblock-stub>
      </anonymous-stub>
    </div>
  </div>
</section>"
`;
